### ISSUE TITLE
React 18 support

### ISSUE THREAD

**Issue by orangecoding** [airbnb/react-dates#2199]
**react-dates version**
latest

**Describe the bug**
React-Dates is using deprecated functions throughout the whole code base like `componentWillReceiveProps`. This throws lot's of errors like these.

```
Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 18.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.
```

React-Dates not being updated is probably blocking lot's of folks to upgrade to the latest React version. What is the plan for this?

**Comment by ljharb** [comment]
See #2105, #1748.

**Comment by TravnikovDev** [comment]
@ljharb maybe you saw a fork which supports modern technologies?

**Comment by ljharb** [comment]
@TravnikovDev im not aware of any forks nor would i recommend them, and your use of the word “modern” here is unhelpful.

**Comment by ljharb** [comment]
It's definitely planned, so this should stay open.

**Comment by Elegant-Bird** [comment]
Any updates on when this is slated for release?

**Comment by ljharb** [comment]
@Elegant-Bird nope, given that react 17 support isn't even working, merged, or released yet.

Like everywhere on github, always, if there's updates, they'll be posted on the issue.

**Comment by suenot** [comment]
Any updates or hacks?

**Comment by aerialist7** [comment]
Due to the fact that `react-dates` library is not supported I forked it, then released and publish a new version that includes this fix.
Anyone can find it by the next npm coordinates: [`"@aerialist7/react-dates": "21.8.0-patch.0"`](https://www.npmjs.com/package/@aerialist7/react-dates/v/21.8.0-patch.0)

**Comment by eduardoacskimlinks** [comment]
Do you have any update on the support for React 18? 

I found even a [PR](https://github.com/react-dates/react-dates/pull/2235) was to upgrade this, but it's closed

**Comment by eduardoacskimlinks** [comment]
From what I read across #2105 #1748, it comes across like there needs to be more interest in maintaining up to date the project. Could you confirm if this package is no longer maintained? 

In case we need to start the transition to another tool. Thanks in advance

**Comment by ljharb** [comment]
The package remains maintained, but that doesnt mean I’ve had time to work on react 17 and 18 support.

Companies donating employee time and/or funds to cover my own time are what would help with that (something that would have helped as far back as 4-5 years ago, just to illustrate how long there’s been effectively zero help)

**Comment by orangecoding** [comment]
Isn't that a lib build and supported by Airbnb?!

**Comment by ljharb** [comment]
@orangecoding airbnb hasn’t contributed anything to react-dates or enzyme in many years.

**Comment by aerialist7** [comment]
I've added React 18 support to my fork.
Anyone can find it on npm by next coordinates: `@aerialist7/react-dates@21.8.0-patch.2`

**Comment by phazei** [comment]
> The package remains maintained, but that doesnt mean I’ve had time to work on react 17 and 18 support.
> 
> Companies donating employee time and/or funds to cover my own time are what would help with that (something that would have helped as far back as 4-5 years ago, just to illustrate how long there’s been effectively zero help)

Maintained: 4 years no new release 😂

**Comment by ljharb** [comment]
@phazei yep! "maintained" doesn't require releases on any particular schedule.

**Comment by eduardoacskimlinks** [comment]
> The package remains maintained, but that doesnt mean I've had time to work on react 17 and 18 support.
> 
> Companies donating employee time and/or funds to cover my own time are what would help with that (something that would have helped as far back as 4-5 years ago, just to illustrate how long there's been effectively zero help)

Thanks for all your years of effort on this project. It's been instrumental, and I understand your decision. I hope we can help you in the future. While I cannot provide immediate support personally, I will make sure to keep this in mind. You've done a fantastic job with this package for so long.

**Comment by corydeppen** [comment]
With the release of React 19 on the horizon, I'm seeing the following after upgrading to React 18.3.1:

`Warning: DateRangePickerInput: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.`

Based on the prior comments about supporting versions 17 and 18, I'm trying to get an idea what the future holds for this project and whether we should be evaluating alternatives if it's possible the latest version of React will not be supported.

**Comment by ljharb** [comment]
@corydeppen once enzyme supports a react version, this project will - which includes react 19.

enzyme remains open for sponsorship and pairing opportunities (going on 7 years now) for any companies that want that support to land faster.

**Comment by aerialist7** [comment]
> With the release of React 19 on the horizon, I'm seeing the following after upgrading to React 18.3.1:
> 
> `Warning: DateRangePickerInput: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.`
> 
> Based on the prior comments about supporting versions 17 and 18, I'm trying to get an idea what the future holds for this project and whether we should be evaluating alternatives if it's possible the latest version of React will not be supported.



> @corydeppen once enzyme supports a react version, this project will - which includes react 19.
> 
> enzyme remains open for sponsorship and pairing opportunities (going on 7 years now) for any companies that want that support to land faster.


@corydeppen, @ljharb if you are interested in support for React 19, then you can try my fork where I supported it and fixed `Support for defaultProps will be removed from function components in a future major release` warning.

It was released in [`@aerialist7/react-dates@21.8.0-patch.3`](https://www.npmjs.com/package/@aerialist7/react-dates/v/21.8.0-patch.3)
Repo: https://github.com/aerialist7/react-dates

**Comment by phazei** [comment]
Project officially dead!  😵🎉
